food-truck-api/src/main/java/food/truck/api/notification/NotificationService.java

```diff
@@ -0,0 +1,102 @@
+package food.truck.api.notification;
+
+import food.truck.api.reviews_and_subscriptions.Subscription;
+import food.truck.api.reviews_and_subscriptions.SubscriptionRepository;
+import food.truck.api.truck.Truck;
+import food.truck.api.user.User;
+import lombok.extern.log4j.Log4j2;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.mail.MailException;
+import org.springframework.mail.javamail.JavaMailSender;
+import org.springframework.stereotype.Service;
+
+import javax.mail.MessagingException;
+import javax.mail.internet.MimeMessage;
+import java.time.Instant;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Log4j2
+@Service
+public class NotificationService {
+    @Autowired
+    NotificationRepository notificationRepository;
+
+    @Autowired
+    SubscriptionRepository subscriptionRepository;
+
+    @Autowired
+    JavaMailSender mailSender;
+
+    public Optional<Notification> findById(long id) {
+        return notificationRepository.findById(id);
+    }
+
+    public List<NotificationView> findBySubscriptionId(long subId) {
+        return notificationRepository.findBySubscriptionId(subId)
+            .stream()
+            .map(NotificationView::of)
+            .collect(Collectors.toList());
+    }
+
+    public List<NotificationView> findNotificationsByUser(User user) {
+        var subscriptions = subscriptionRepository.findByUser(user);
+        return subscriptions.stream()
+            .flatMap(sub -> findNotificationsBySubscriptionId(sub.getId()).stream())
+            .map(NotificationView::of)
+            .collect(Collectors.toList());
+    }
+
+    public List<NotificationView> findNotificationsByTruck(Truck truck) {
+        var subscriptions = subscriptionRepository.findByTruck(truck);
+        return subscriptions.stream()
+            .flatMap(sub -> findNotificationsBySubscriptionId(sub.getId()).stream())
```

Comment:
Are we certain that we're only going to have truck owners able to communicate with subscribed members? I thought there was going to be a 'general broadcast' ability to anyone within their radius. Or is that accounted for elsewhere in the code?

Reply:
This has not been accounted for yet. I was focusing on notifying subscribers for this scrum deliverable.